db/migrate/20171002152414_fix_dfid_research_output_first_published_at.rb

```diff
@@ -0,0 +1,18 @@
+class FixDfidResearchOutputFirstPublishedAt < ActiveRecord::Migration[5.1]
+  # There are ~63000 editions which have a first_published_at timestamp
+  # later than their public_updated_at timestamp. So use legacy data from
+  # metadata to fix this.
+  #
+  def up
+    sql = <<-SQL.strip_heredoc
+      UPDATE editions
+      SET    first_published_at = (details->'metadata'->>'first_published_at')::date
+      WHERE  publishing_app = 'specialist-publisher'
+      AND    document_type  = 'dfid_research_output'
+      AND    first_published_at > (public_updated_at + interval '1 second')
```

Comment:
Why do we need the 1 second interval here?

Reply:
I found that an edition could have the same datetime to the second or even millisecond for both fields but the storage to microsecond precision meant that they were not deemed equal when executing the query.
So we'd get a lot of legitimately near-equal datetimes being caught by what is essentially a search for nonsensical date combinations. In other words rounding.

Reply:
Ah yep, makes sense.

On a side note, I'll try and figure out why the e2e tests are failing...